receiver/netflowreceiver/producer.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package netflowreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/netflowreceiver"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/netsampler/goflow2/v2/producer"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/netflowreceiver/internal/metadata"
+)
+
+// OtelLogsProducerWrapper is a wrapper around a producer.ProducerInterface that sends the messages to a log consumer
+type OtelLogsProducerWrapper struct {
+	wrapped     producer.ProducerInterface
+	logConsumer consumer.Logs
+	logger      *zap.Logger
+}
+
+var ErrorProducerProcessing = fmt.Errorf("unexpected error processing the message")
+
+// Produce converts the message into a list log records and sends them to log consumer
+func (o *OtelLogsProducerWrapper) Produce(msg any, args *producer.ProduceArgs) ([]producer.ProducerMessage, error) {
+	if pErr := recover(); pErr != nil {
+		errMessage, _ := pErr.(string)
+		o.logger.Error("unexpected error processing the message", zap.String("error", errMessage))
+		return nil, ErrorProducerProcessing
+	}
```

Comment:
```suggestion
	defer func() {
		if pErr := recover(); pErr != nil {
			errMessage, _ := pErr.(string)
			o.logger.Error("unexpected error processing the message", zap.String("error", errMessage))
			return nil, ErrorProducerProcessing
		}
	}
```
The panic recovery has to happen in a deferred function execution to work. Also, is there a way we can test this?

Reply:
Sorry I missed this obvious mistake, writing tests for it, I think we can create a producer that panics, and use it in place of the proto producer on the test

The only panic here would come from the proto producer implementation

Reply:
Added it [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36865/commits/af1e72bd75a26de470fba6bfd737a4a5625c2f29)